### PR TITLE
Update heroku.md with new considerations

### PR DESCRIPTION
**Pull request by chaNcharge**
I also added my script that I used to keep FredBoat awake by pinging it every 15 minutes. Could you also review my changes and script in case I missed something?

EDIT: I created this script because some people might want it only on during certain times of the day. By having it on 17 hours for me, it doesn't eat up all the free dyno hours without a credit card.

### CODE REVIEW

File: public/markdown/heroku.md [hunk 1]
```diff
@@ -95,7 +95,8 @@ git push heroku master
 ### Considerations
 Hosting FredBoat this way is fine, but there are some downsides using Heroku to host FredBoat.
 
-1. On the free and hobby plans, Heroku allows a maximum of 24 hours uptime, with a delay up to 2 to 3 hours. This means that your FredBoat restarts around the time you deployed the code. This can differ up to 4 hours.
+1. On the free and hobby plans, Heroku allows a maximum of 24 hours uptime, with a delay up to 2 to 3 hours. This means that your FredBoat restarts around the time you deployed the code. This can differ up to 4 hours.
```
Comment (Macley-Kun) [inline public/markdown/heroku.md]: Think a little type error... Extra space :P

File: heroku_keepawake.gs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+function pingApp() {
```
Comment (Macley-Kun) [inline heroku_keepawake.gs]: I'm not a wizard. But is it possible to make this more user friendly? If not then ig it's fine for the onces who are looking for this.

File: public/markdown/heroku.md [hunk 3]
```diff
@@ -95,7 +95,8 @@ git push heroku master
...
+4. You only get 550 hours of uptime per month on the free plans without a credit card. If you verify your Heroku account with a credit card, you get 1000 hours per month, which is more than enough to keep FredBoat awake 24/7.
+5. On the free plan, Heroku will put FredBoat to sleep after 30 minutes of inactivity. There's many different ways to prevent Heroku from putting FredBoat to sleep. The easiest way of keeping the bot awake is to create a Google Apps Script. You can create your own or use this [script](https://github.com/chaNcharge/fredboat.com/blob/master/heroku_keepawake.gs) and add it to a new Google Sheet. If you use this script, make sure to read the comments and add your Heroku App URL and Sheet ID (found in the URL of your Sheet). The advantage to using this script is that it keeps FredBoat on for 17 hours a day, which complies with the consideration above.
```
Comment (Macley-Kun) [inline public/markdown/heroku.md]: I would suggest users to use this: http://kaffeine.herokuapp.com/
If they really need your script, i would then suggest them using that.

**Comment by Macley-Kun** [comment]
Looks good :-)

Really completes the whole thing! Users withoud a cc can now host a fred too :^)

I only wonder if the http://kaffeine.herokuapp.com/ should be a real step in the instruction. Because withoud it, we technically don't have a working bot hehe..

Up to you to decide :^) i'm already happy with the changes!

**Comment by chaNcharge** [comment]
Alright, I made it more user friendly by adding more comments. I think it looks good as it is! If there's other problems I should look at, let me know.

**Comment by Macley-Kun** [comment]
Looks good :^) 
People should be able to use both solutions.

**Comment by Macley-Kun** [comment]
Is it an idea to make the use of https://kaffeine.herokuapp.com/ a necessary step? I've bin thinking about this consideration, but if users wouldn't use any solution to keep it awake, it'll sleep.

**Comment by chaNcharge** [comment]
I think you're right that it should be a necessary step, I'll try editing the steps a bit.

**Comment by Macley-Kun** [comment]
Fred the pull is ready! Should fix the steps that i looked over!

**Comment by chaNcharge** [comment]
I'm still testing the script for errors, it should be 100% ready in a week.

**Comment by chaNcharge** [comment]
@Frederikam I am confident that it is good to go now. Go ahead and merge it.

**Comment by Macley-Kun** [comment]
Make sure you're working in a different folder for this.
Make sure you're working in a new folder for this (not inside the one we just worked in).

Could you change that for me?
image
^couldn't blame the guy, wasn't clear enough.

**Comment by chaNcharge** [comment]
Wait, is the cache folder not supposed to be in the main app folder? I made mine like that and it worked just fine:
screen shot 2018-06-09 at 12 57 47 pm

File: public/markdown/heroku.md [hunk 4]
```diff
@@ -97,5 +100,6 @@ Hosting FredBoat this way is fine, but there are some downsides using Heroku to
...
 2. Heroku uses an ephemeral filesystem. This means that every change that occurs after you pushed your files isn't saved. The downside of this is that when FredBoat restarts, no songs are reloaded. Also, all the errors that aren't sent to the Sentry are lost too.
-3. It's possible to use the local file playing, it's limited however due to the maximum file size of 500mb per app. The project itself uses up to 200mb. You can experience slower boot times when exceeding 400mb.
-4. Heroku will put FredBoat to sleep after 30 minutes of inactivity. There's many different ways to prevent Heroku from putting your bot to sleep. You can create a script that pings the bot every now and then to keep it awake, but keep in mind that it will use up more of your hours of your plan. You can also try using the Heroku Scheduler addon to keep it awake during certain times of the day. You can read about it [here](https://devcenter.heroku.com/articles/free-dyno-hours)
+3. It's possible to play local files; however, it's limited due to the maximum file size of 500mb per app. The project itself uses up to 200mb. You can experience slower boot times when exceeding 400mb.
```
Comment (Frederikam) [inline public/markdown/heroku.md]: Are you sure it's Mb or MB? 1 MB = 8Mb

**Comment by Frederikam** [comment]
I think I would move the procfile and the google script to the public directory. That way we can link them from the fredboat.com domain

File: public/markdown/heroku.md (hunk 4 above)
Comment (Macley-Kun) [reply public/markdown/heroku.md]: 500 MiB

They use MiB's, so.... change all mb to MiB 😛

**Comment by chaNcharge** [comment]
I moved the files, but how should I link them in the markdown file?

**Comment by Frederikam** [comment]
They should be linked like `/Procfile` etc

**Comment by chaNcharge** [comment]
Oh, so not on github, but on fredboat.com?

**Comment by chaNcharge** [comment]
I changed the links, but it seems the deploy preview breaks again.

**Comment by Frederikam** [comment]
Problems with the NPM cache again. I cleared it

**Comment by Frederikam** [comment]
@chaNcharge you're overcomplicating things. Format the links like `/procfile`, not `https://fredboat.com/procfile`. The former is relative to the domain, which in some cases is not fredboat.com.

**Pull request merged**